Failing TestConversionURLValues

When I run tests with `-shuffle=on`, the `TestConversionURLValues` test sometimes fails.

It can be reproduced with:

```console
❯ go version
go version go1.26.0 darwin/arm64
❯ go test -shuffle=on -run=^TestConversion -test.shuffle 1771956053236831000 ./...
-test.shuffle 1771956053236831000
--- FAIL: TestConversionURLValues (0.00s)
    conversions_test.go:98: not equal:
        expected: url.Values{"abc":[]string{"123"}, "bools[]":[]string{"true", "false"}, "data[age]":[]string{"30"}, "data[arr][]":[]string{"1", "2"}, "data[height]":[]string{"162"}, "mapSlice[][age]":[]string{"40"}, "mapSlice[][height]":[]string{"152"}, "msiData[age]":[]string{"30"}, "msiData[arr][]":[]string{"1", "2"}, "msiData[height]":[]string{"162"}, "msiSlice[][age]":[]string{"40"}, "msiSlice[][height]":[]string{"152"}, "name":[]string{"Mat"}, "stats[]":[]string{"1", "2"}}
        actual:   url.Values{"abc":[]string{"123"}, "bools":[]string{"true", "false"}, "data[age]":[]string{"30"}, "data[arr]":[]string{"1", "2"}, "data[height]":[]string{"162"}, "mapSlice[age]":[]string{"40"}, "mapSlice[height]":[]string{"152"}, "msiData[age]":[]string{"30"}, "msiData[arr]":[]string{"1", "2"}, "msiData[height]":[]string{"162"}, "msiSlice[age]":[]string{"40"}, "msiSlice[height]":[]string{"152"}, "name":[]string{"Mat"}, "stats":[]string{"1", "2"}}
--- FAIL: TestConversionURLQuery (0.00s)
    conversions_test.go:127: not equal:
        expected: "abc=123&bools[]=true&bools[]=false&data[age]=30&data[arr][]=1&data[arr][]=2&data[height]=162&mapSlice[][age]=40&mapSlice[][height]=152&msiData[age]=30&msiData[arr][]=1&msiData[arr][]=2&msiData[height]=162&msiSlice[][age]=40&msiSlice[][height]=152&name=Mat&stats[]=1&stats[]=2"
        actual:   "abc=123&bools=true&bools=false&data[age]=30&data[arr]=1&data[arr]=2&data[height]=162&mapSlice[age]=40&mapSlice[height]=152&msiData[age]=30&msiData[arr]=1&msiData[arr]=2&msiData[height]=162&msiSlice[age]=40&msiSlice[height]=152&name=Mat&stats=1&stats=2"
FAIL
FAIL    github.com/stretchr/objx        0.218s
FAIL
```

--- a/conversions_test.go
+++ b/conversions_test.go
@@ -128,6 +128,7 @@ func TestConversionURLQuery(t *testing.T) {
 }
 
 func TestConversionURLQueryNoSliceKeySuffix(t *testing.T) {
+	t.Cleanup(func() { _ = objx.SetURLValuesSliceKeySuffix(objx.URLValuesSliceKeySuffixArray) })
 	m := getURLQueryMap()
 	err := objx.SetURLValuesSliceKeySuffix(objx.URLValuesSliceKeySuffixEmpty)
 	require.Nil(t, err)
@@ -144,6 +145,7 @@ func TestConversionURLQueryNoSliceKeySuffix(t *testing.T) {
 }
 
 func TestConversionURLQueryIndexSliceKeySuffix(t *testing.T) {
+	t.Cleanup(func() { _ = objx.SetURLValuesSliceKeySuffix(objx.URLValuesSliceKeySuffixArray) })
 	m := getURLQueryMap()
 	m.Set("mapSlice", []objx.Map{{"age": 40, "sex": "male"}, {"height": 152}})
 	err := objx.SetURLValuesSliceKeySuffix(objx.URLValuesSliceKeySuffixIndex)
@@ -161,6 +163,7 @@ func TestConversionURLQueryIndexSliceKeySuffix(t *testing.T) {
 }
 
 func TestValidityURLQuerySliceKeySuffix(t *testing.T) {
+	t.Cleanup(func() { _ = objx.SetURLValuesSliceKeySuffix(objx.URLValuesSliceKeySuffixArray) })
 	err := objx.SetURLValuesSliceKeySuffix("")
 	assert.Nil(t, err)
 	err = objx.SetURLValuesSliceKeySuffix("[]")

